magma/backend/coreir/coreir_transformer.py

```diff
@@ -37,6 +39,45 @@
 _generator_callbacks = {}
 
 
+def _check_wiring_context(i, o):
```

Comment:
I think ideally this should be called during a checking pass, independent of CoreIR, i.e. before lowering to CoreIR. But I can see performance implications for that being a separate pass and having to iterate over all the connections multiple times, so need to think about it for a bit.

At least for now, can we move this function into a separate file? Maybe `magma/backend/check_wiring_context.py: check_wiring_context`.

Reply:
To make it modular while avoiding the performance implications, we could provide it somewhere in the class ancestry as part of the super logic.  Maybe as part of an API to "get the driver" of a value (since we need the source to wire up the value/sink).  Then as part of the API, we run the error check and this can be reused by backends.